app/backend/approaches/chatreadretrieveread.py

```diff
@@ -80,6 +81,23 @@ async def run_until_final_call(
 
         user_q = "Generate search query for: " + history[-1]["user"]
```

Comment:
This variable is confusingly named now, since there's also a user_query parameter in the new function that's only the latter part of this variable. Maybe this can be "user_query_request" or something. Or it might be clearer to store both vars here:

```
user_query = history[-1]["user"]
user_query_request = f"Generate search query for: {user_query}"
```

And then pass those around as needed to the functions below.